packages/salesforcedx-vscode-core/src/commands/util/commandWrapper.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { CommandEventStream, CommandEventType } from '@salesforce/salesforcedx-utils-vscode';
+import * as vscode from 'vscode';
+
+export const registerCommand = (commandId: string, callback: (...args: any[]) => any, thisArg?: any): vscode.Disposable => {
+  return vscode.commands.registerCommand(commandId, wrapCommandCallback(commandId, callback), thisArg);
+};
+
+const wrapCommandCallback = (commandId: string, callback: (...args: any[]) => any): (...args: any[]) => any => {
+  return async (...args: any[]) => {
+    CommandEventStream.getInstance().post({ type: CommandEventType.START, commandId });
+    await callback(...args);
```

Comment:
I think we need to return the result of this.  One thing I'm not clear about...what if the caller doesn't want to wait on the promise to complete.  

Should this be something like: 
```
...
const result = Promise.resolve(callback(...args)).then((result) => {
  CommandEventStream.getInstance().post({ type: CommandEventType.END, commandId });
  return result;
});
```
That way we always get the END event recorded, but the caller has control on awaiting the promise, and access to the result of the call when desired.